lib/vfscore/automount.c

```diff
@@ -178,11 +225,20 @@ static int vfscore_volume_process_ukopts(struct vfscore_volume *vv)
 
 static inline int vfscore_mount_volume(struct vfscore_volume *vv)
 {
+	UK_ASSERT(vv);
+
 #if CONFIG_LIBVFSCORE_FSTAB
 	vfscore_volume_process_ukopts(vv);
 #endif /* CONFIG_LIBVFSCORE_FSTAB */
 
-	return mount(vv->sdev, vv->path, vv->drv, vv->flags, vv->opts);
+#if CONFIG_LIBUKCPIO && CONFIG_LIBRAMFS
+	if (!strncmp(vv->drv, "initrd", sizeof("initrd") - 1)) {
```

Comment:
Using `strcmp` should be safe because the const string `"initrd"` is `'\0'`-terminated. The check here excludes the `\0`-termination symbol, this is why it matches with everything `initrd*` although it should stop matching with `initrd`.